src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/PreloginSubHandlers/PreloginHandlerContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClient.SNI;
+
+namespace Microsoft.Data.SqlClientX.Handlers.Connection.PreloginSubHandlers
+{
+    /// <summary>
+    /// Handler context for Prelogin.
+    /// </summary>
+    internal class PreLoginHandlerContext : HandlerRequest
+    {
+        public SqlConnectionEncryptOption ConnectionEncryptionOption { get; private set; }
+        public bool IsTlsFirst { get; private set; }
+        public bool TrustServerCert { get; private set; }
+        public bool IntegratedSecurity { get; private set; }
+        public SqlAuthenticationMethod AuthType { get; private set; }
+        public string HostNameInCertificate { get; private set; }
+        public string ServerCertificateFilename { get; private set; }
+
+        public EncryptionOptions InternalEncryptionOption { get; set; } = EncryptionOptions.OFF;
+
+        public ConnectionHandlerContext ConnectionContext { get; private set; }
+        public bool ValidateCertificate { get; internal set; }
+        public SNIError SniError { get; internal set; }
+        public bool ServerSupportsEncryption { get; internal set; }
+        public PreLoginHandshakeStatus HandshakeStatus { get; internal set; }
+
+        public PreLoginHandlerContext(ConnectionHandlerContext connectionContext)
+        {
+            ConnectionContext = connectionContext;
+            SqlConnectionString connectionOptions = connectionContext.ConnectionString;
+            ConnectionEncryptionOption = connectionOptions.Encrypt;
+            IsTlsFirst = (ConnectionEncryptionOption == SqlConnectionEncryptOption.Strict);
```

Comment:
Is this better as a calculated property?

Reply:
Yeah, a convenience being offered, so that we dont check for the ConnectionEncryptionOption to figure out if Tls needs to be done before Prelogin or after exchanging prelogin information.

Reply:
@edwardneal  what did you mean to suggest? Offer this as a Method ? Sorry didn't udnerstand the intention behind the comment.

Reply:
At the moment, you have a property named `IsTlsFirst` and manually set it to the result of a calculation. I was wondering whether it'd be better to define `IsTlsFirst` as the calculation itself:

```csharp
public bool IsTlsFirst
    => (ConnectionEncryptionOption == SqlConnectionEncryptOption.Strict);
```

Reply:
Thanks for explaining. I had completely missed the point.